src/main/java/seedu/address/model/room/UniqueRoomList.java

```diff
@@ -271,11 +277,13 @@ public void clearRoom(Name patientName) {
             Name patientNameInRoom = internalList.get(i - 1).getPatient().get().getName();
             if (patientName.equals(patientNameInRoom)) {
                 Room roomToClear = internalList.get(i - 1);
-                setSingleRoom(roomToClear, new Room(roomToClear.getRoomNumber()));
+                setSingleRoom(roomToClear, new Room(roomToClear.getRoomNumber(),
+                        false, Optional.empty(), roomToClear.getRoomTasks()));
```

Comment:
It is possible to use `new RoomTasks(roomToClear.getReadOnlyTasks())` instead to avoid adding a getter!